addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -101,10 +92,8 @@ def _compute_show_validate(self):
     def _compute_state(self):
         batchs = self.filtered(lambda batch: batch.state not in ['cancel', 'done'])
         for batch in batchs:
-            if not batch.picking_ids:
-                return
-            # Cancels automatically the batch picking if all its transfers are cancelled.
-            if all(picking.state == 'cancel' for picking in batch.picking_ids):
+            # Cancels the batch picking if no picking or all its transfers are cancelled.
+            if not batch.picking_ids or all(picking.state == 'cancel' for picking in batch.picking_ids):
```

Comment:
I'm not sure it's a good idea anymore, because if you create a new batch then save it without add any pickings, it will be cancelled immediately. It doesn't seems dramatic but really strange.
So maybe we need to filter on in progress batches only.
But in the other hand, if we do that, a batch emptied by moving its pickings into an another batch will not be cancelled if it was still in `draft`...

I also tried to use `_origin` to see if the batch had pickings previously, but it doesn't seem to work.

Reply:
@svs-odoo moved this part to picking `write`, can you check it?